File: python/caesar_cipher.py
def caesar_cipher(string, shift_amount):
    result = []
    for i in range(len(string)):
        code = ord(string[i])
        cipher_code = ord(string[i]) + shift_amount
        is_char = string[i].isalpha()
        if string[i].upper() == string[i] and is_char:
            if cipher_code > 90:
                cipher_code = (cipher_code - 26)
            elif cipher_code < 65:
                cipher_code = (cipher_code + 26)
            result.append(chr(cipher_code))
        elif string[i].lower() == string[i] and is_char:
            if cipher_code > 122:
                cipher_code = (cipher_code - 26)
            elif cipher_code < 97:
                cipher_code = (cipher_code + 26)
            result.append(chr(cipher_code))
        else:
            result.append(string[i])
            
    return "".join(result)

File: js/caesarCipher.js
exports.caesarCipher = function(str, num) {
    let result = []
    for(let i = 0; i < str.length; i++) {
        let code = str.charCodeAt(i)
        let cipherCode = str.charCodeAt(i) + num
        let isChar = (/[a-zA-Z]/).test(str[i])
        if (str[i].toUpperCase() === str[i] && isChar) {
            if(cipherCode > 90){
                cipherCode = (cipherCode - 26)
            } else if(cipherCode < 65){
                cipherCode = (cipherCode + 26)
            }
            result.push(String.fromCharCode(cipherCode))
        } else if (str[i].toLowerCase() === str[i] && isChar){
            if(cipherCode > 122){
                cipherCode = (cipherCode - 26)
            } else if(cipherCode < 97){
                cipherCode = (cipherCode + 26)
            }
            result.push(String.fromCharCode(cipherCode))
        } else {
            result.push(str[i])
        }
    }
    
    return result.join("")
}

